src/resmom/mom_hook_func.c

```diff
@@ -3075,6 +3077,313 @@ record_job_last_hook_executed(unsigned int hook_event,
 	fclose(fp);
 }
 
+ static int run_execjob_end_hooks(struct work_task *);
+
+/**
+ * @brief
+ * This function runs after execution of a single execjob_end hook and
+ * processes the results from hook execution. On successful hook
+ * execution, a new task will be created to run the next hook script.
+ * If there was an error (the hook process returned a non-zero exit
+ * status) it does not create the new task for the next hook script
+ * and replies to the outstanding batch request.
+ * 
+ * @param[in] 	ptask - the work task.
+ *
+ * @return none
+ */
+static void
+post_execjob_end_hook(struct work_task *ptask)
+{
+
+	int    accept_flag = 1;
+	int    reject_flag = 0;
+	int    reject_rerunjob = 0;
+	int    reject_deletejob = 0;
+	int    reboot_flag = 0;
+	int    log_type = 0;
+	int    log_class = 0;
+	int    hook_error_flag = 0;
+	int    wstat = 0;
+	char   *log_id = NULL;
+	char   reject_msg[HOOK_MSG_SIZE+1] = {'\0'};
+	char   reboot_cmd[HOOK_BUF_SIZE]  = {'\0'};
+	char   hook_outfile[MAXPATHLEN+1] = {'\0'};
+	pid_t  mypid = 0;
+	hook   *phook	= NULL;
+	mom_hook_input_t *hook_input 	= NULL;
+	job *pjob	= NULL;
+	struct work_task *new_task;
+	pbs_list_head	vnl_changes;
+#if !MOM_ALPS
+	struct batch_request *preq;
+	int 	numnodes = 0;
+#endif
+
+	if (ptask == NULL) {
+		log_err(-1, __func__, "missing ptask argument to event");
+		return;
+	}
+
+	hook_input = (mom_hook_input_t *) ptask->wt_parm2;
```

Comment:
I don't see run_hook() setting wt_parm2, All I see is that run_hook() is setting hook pointer in wt_parm1. So wouldn't this always fail?

Reply:
I am setting wt_param2 in run_hook() at [#L805](https://github.com/PBSPro/pbspro/blob/f2da3a05753ca04e8577528ac742180a7cd9ebd3/src/resmom/mom_hook_func.c#L805).